packages/vector_graphics/lib/src/listener.dart

```diff
@@ -48,28 +55,70 @@ Future<PictureInfo> decodeVectorGraphics(
   required BytesLoader loader,
 }) {
   try {
+    // We might be in a test that's using a fake async zone. Make sure that any
+    // real async work gets scheduled in the root zone so that it will not get
+    // blocked by microtasks in the fake async zone, but do not unnecessarily
+    // create zones outside of tests.
+    bool useZone = false;
     assert(() {
       _debugLastTextDirection = textDirection;
       _debugLastLocale = locale;
+      useZone = Zone.current != Zone.root &&
+          Zone.current.scheduleMicrotask != Zone.root.scheduleMicrotask;
       return true;
     }());
-    final FlutterVectorGraphicsListener listener =
-        FlutterVectorGraphicsListener(
-      locale: locale,
-      textDirection: textDirection,
-      clipViewbox: clipViewbox,
-    );
-    DecodeResponse response = _codec.decode(data, listener);
-    if (response.complete) {
-      return SynchronousFuture<PictureInfo>(listener.toPicture());
+
+    @pragma('vm:prefer-inline')
+    Future<PictureInfo> process() {
+      final FlutterVectorGraphicsListener listener =
+          FlutterVectorGraphicsListener(
+        locale: locale,
+        textDirection: textDirection,
+        clipViewbox: clipViewbox,
+      );
+      DecodeResponse response = _codec.decode(data, listener);
+      if (response.complete) {
+        return SynchronousFuture<PictureInfo>(listener.toPicture());
+      }
+      assert(() {
+        _pendingDecodes[loader] = Completer<void>();
+        return true;
+      }());
+      return listener.waitForImageDecode().then((_) {
+        response = _codec.decode(data, listener, response: response);
+        assert(response.complete);
+        assert(() {
+          _pendingDecodes.remove(loader)?.complete();
+          return true;
+        }());
+        return listener.toPicture();
+      });
+    }
+
+    if (!useZone) {
```

Comment:
I think if we only need this in debug mode, you could re-arrange this a bit to make the tree shaker work a bit better:

```
if (!kDebugMode || !useZone) return process
```

Reply:
done